transports/httpclient/src/test/java/org/commonjava/maven/galley/transport/htcli/ContentFilteringTransferDecoratorTest.java

```diff
@@ -0,0 +1,188 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.maven.galley.transport.htcli;
+
+import com.codahale.metrics.MetricRegistry;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.commonjava.maven.galley.config.TransportMetricConfig;
+import org.commonjava.maven.galley.event.EventMetadata;
+import org.commonjava.maven.galley.io.OverriddenBooleanValue;
+import org.commonjava.maven.galley.model.ConcreteResource;
+import org.commonjava.maven.galley.model.Location;
+import org.commonjava.maven.galley.model.Transfer;
+import org.commonjava.maven.galley.spi.transport.DownloadJob;
+import org.commonjava.maven.galley.transport.htcli.internal.HttpDownload;
+import org.commonjava.maven.galley.transport.htcli.model.SimpleHttpLocation;
+import org.commonjava.maven.galley.transport.htcli.testutil.HttpTestFixture;
+import org.commonjava.test.http.expect.ExpectationHandler;
+import org.hamcrest.CoreMatchers;
+import org.junit.Rule;
+import org.junit.Test;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+public class ContentFilteringTransferDecoratorTest
+{
+    @Rule
+    public HttpTestFixture fixture = new HttpTestFixture( "test" );
+
+    private static MetricRegistry metricRegistry = new MetricRegistry();
+
+    private static TransportMetricConfig metricConfig = new TransportMetricConfig()
+    {
+        @Override
+        public boolean isEnabled()
+        {
+            return true;
+        }
+
+        @Override
+        public String getNodePrefix()
+        {
+            return null;
+        }
+
+        @Override
+        public String getMetricUniqueName( Location location )
+        {
+            if ( location.getName().equals( "test" ) )
+            {
+                return location.getName();
+            }
+            return null;
+        }
+    };
+
+    @Test
+    public void snapshotNotExistsWhenSnapshotsNotAllowed()
+            throws Exception
+    {
+        final String fname = "/commons-codec/commons-codec/11-SNAPSHOT/maven-metadata.xml.md5";
+
+        final String content = "kljsjdlfkjsdlkj123j13=20=s0dfjklxjkj";
+        fixture.getServer().expect( "GET", fixture.formatUrl( fname ), new ExpectationHandler()
```

Comment:
There's a variant of this method that just accepts a string body and response status code, so you don't have to implement a handler. Just FYI